src/Symfony/Contracts/Translation/TranslatorInterface.php

```diff
@@ -52,14 +52,14 @@ interface TranslatorInterface
      *
      * @see https://en.wikipedia.org/wiki/ISO_31-11
      *
-     * @param string      $id         The message id (may also be an object that can be cast to string)
-     * @param array       $parameters An array of parameters for the message
-     * @param string|null $domain     The domain for the message or null to use the default
-     * @param string|null $locale     The locale or null to use the default
+     * @param string|\Symfony\Contracts\Translation\TranslatableInterface $id         The message id (may also be an object that can be cast to string)
+     * @param array                                                       $parameters An array of parameters for the message
+     * @param string|null                                                 $domain     The domain for the message or null to use the default
+     * @param string|null                                                 $locale     The locale or null to use the default
      *
      * @return string The translated string
      *
      * @throws \InvalidArgumentException If the locale contains invalid characters
      */
-    public function trans(string $id, array $parameters = [], string $domain = null, string $locale = null);
+    public function trans($id, array $parameters = [], string $domain = null, string $locale = null);
```

Comment:
we may also aim for a `string|Stringable` notation.

Reply:
If we do this, wouldn't that insinuate that this object should also implement `Stringable`? In order for the object to be converted to a string, it would need to be aware of the translator and wouldn't be a simple data object.

Reply:
yeah, never mind :) it only would only provide us the message ID (__toString), but not parameters and such.